File: backend/app/schemas/tip.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class TipCreate(BaseModel):
    amount: int
    name: Optional[str] = None
    message: Optional[str] = None
    isPrivate: bool = False
    creator_profile_id: int
    stripe_session_id: Optional[str] = None

class TipOut(BaseModel):
    id: int
    amount: int
    name: Optional[str]
    message: Optional[str]
    isPrivate: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/tip.ts
export interface Tip {
    id: number;
    amount: number;
    name?: string | null;
    message?: string | null;
    isPrivate: boolean;
    created_at: Date;
}

export interface GetTipResponse {
  tips: Tip[];
}

export interface GetTipRequest {
    creator_profile_id: number;
    limit: number;
    offset: number;
}
